src/flamegpu/io/Telemetry.cpp

```diff
@@ -0,0 +1,167 @@
+#include "flamegpu/io/Telemetry.h"
+
+#include <cstdlib>
+#include <cerrno>
+#include <algorithm>
+#include <memory>
+#include <array>
+#include <iostream>
+#include <cstdio>
+#include <cctype>
+
+
+#include "flamegpu/version.h"
+#include "flamegpu/exception/FLAMEGPUException.h"
+#include "flamegpu/util/Environment.h"
+
+
+
+namespace flamegpu {
+namespace io {
+namespace Telemetry {
+
+const char TELEMETRY_ENDPOINT[] = "https://nom.telemetrydeck.com/v1/";
+const char TELEMETRY_APP_ID[] = "94AC5E3F-F674-4E29-BF87-DAF4BA7F8F79";
+
+std::string generateTelemetryData(std::string event_name, std::map<std::string, std::string> payload_items) {
+    const std::string var_testmode = "$TEST_MODE";
+    const std::string var_appID = "$APP_ID";
+    const std::string var_buildHash = "$BUILD_HASH";
+    const std::string var_eventName = "$EVENT_TYPE";
+    const std::string var_payload = "$PAYLOAD";
+
+    // check ENV for test variable FLAMEGPU_TEST_ENVIRONMENT
+    std::string testmode;
+    if (flamegpu::util::isTestEnvironment())
+        testmode = "true";
+    else
+        testmode = "false";
+    std::string appID = TELEMETRY_APP_ID;
+    std::string buildHash = flamegpu::BUILD_HASH;
+
+    // add app version by checking to see if python run
+    if (flamegpu::util::hasEnvironmentVariable("FLAMEGPU_PYFLAMEGPU_VERSION")) {
+        std::string py_version = "pyflamegpu" + std::string(flamegpu::VERSION_STRING);
+        payload_items["appVersion"] = py_version;                                                                       // e.g. 'pyflamegpu2.0.0-alpha.3' (graphed in Telemetry deck)
+        payload_items["appPythonVersionFull"] = flamegpu::util::getEnvironmentVariable("FLAMEGPU_PYFLAMEGPU_VERSION");  // e.g. '2.0.0a3+cuda116'
+    } else {
+        // Not python environment
+        payload_items["appVersion"] = flamegpu::VERSION_STRING;  // e.g. '2.0.0-alpha.3' (graphed in Telemetry deck)
+    }
+    // other version strings
+    payload_items["appVersionFull"] = flamegpu::VERSION_FULL;
+    payload_items["majorSystemVersion"] = std::to_string(flamegpu::VERSION_MAJOR);        // e.g. '2' (graphed in Telemetry deck)
+    std::string major_minor_patch = std::to_string(flamegpu::VERSION_MAJOR) + "." + std::to_string(flamegpu::VERSION_MINOR) + "." + std::to_string(flamegpu::VERSION_PATCH);
+    payload_items["majorMinorSystemVersion"] = major_minor_patch;                         // e.g. '2.0.0' (graphed in Telemetry deck)
+    payload_items["appVersionPatch"] = std::to_string(flamegpu::VERSION_PATCH);
+    payload_items["appVersionPreRelease"] = flamegpu::VERSION_PRERELEASE;
+    payload_items["buildNumber"] = flamegpu::VERSION_BUILDMETADATA;  // e.g. '0553592f' (graphed in Telemetry deck)
+
+    // OS
+#ifdef _WIN32
+    payload_items["operatingSystem"] = "Windows";
+#elif __GNUC__ >= 4
```

Comment:
This will cause other to be set if clang is used as the host compiler. There are also preproc macros defined commonly such as `__linux__` which can be checked. 

Not sure how / why `4` is relevant.

https://blog.kowalczyk.info/article/j/guide-to-predefined-macros-in-c-compilers-gcc-clang-msvc-etc..html